client/src/js/services/grid/GridState.js

```diff
@@ -0,0 +1,57 @@
+angular.module('bhima.services')
+.service('GridStateService', GridStateService);
+
+GridStateService.$inject = [
+  'util', 'appcache', 'NotifyService'
+];
+
+// responsible for -
+// - caching grid state seperately for each grid
+// - hooking into the grid api to apply the default size only when the grid is ready
+// - exposing the methods to save and restore grid state
+function GridStateService(util, AppCache, Notify) {
+  /* @const */
+  var stateCacheKey = 'gridState';
+
+  function StateInstance(gridOptions, moduleCacheKey) {
+    this._cacheKey = moduleCacheKey.concat(stateCacheKey);
+    this._cache = new AppCache(this._cacheKey);
+
+    util.after(gridOptions, 'onRegisterApi', function onRegisterApi(api) {
+      this._gridApi = api;
+
+      this._gridApi.core.on.rowsRendered(null, util.once(function () {
+        this.restoreGridState();
+      }.bind(this)));
+    }.bind(this));
+
+    this.saveGridState = saveGridState.bind(this);
```

Comment:
It would probably be more efficient to attach these to `StateInstance.prototype`.  Is there a reason not to?